### DISCUSSION TITLE
Difference between `Press Key Block` and `Forms Block` for filling value of  input element

### DISCUSSION THREAD

**Discussion by kapsterio** [AutomaApp/automa#1235] [Q&A]
Hey automa guys, I just met automa yesterday, and it's really cool!!!, thanks for making such cool tool.  

I tried to make a workflow to auto fllling value to the text input on my web system, first I used `Press Key Block`  to do the job, but it didn't work when I submit the input value, it submitted an empty value for whatever `Press Key Block` typed. Then I switched to `Forms Block`, and it works as I expected.

So my question is what the difference between `Press Key Block` and `Forms Block` for filling value of  input element, why empty value is submitted when use `Press Key Block`?

**Comment by fmen1** [comment]
The Press Key Block is rarely used in javascript due to various reasons. Best stick with Forms Block.
